### PR TITLE
Add AnthosL4LB resource map

### PR DESCRIPTION
**Pull request by courageJ**

### CODE REVIEW

File: retrieval/resource_map.go [hunk 1]
```diff
@@ -77,6 +77,17 @@ var GCEResourceMap = ResourceMap{
...
+	Type: "anthos_l4lb",
+	LabelMap: map[string]labelTranslation{
+		ProjectIDLabel:                         constValue("project_id"),
+		KubernetesLocationLabel:                constValue("location"),
+		"__meta_kubernetes_service_annotation_anthos_l4lb_kind":          constValue("kind"),
+		"__meta_kubernetes_service_annotation_anthos_l4lb_group_name":          constValue("group_name"),
+		"__meta_kubernetes_service_annotation_anthos_l4lb_hostname":           constValue("hostname"),
+	},
+}
+
```
Comment (igorpeshansky) [inline retrieval/resource_map.go:90]: Any reason you added this here, and not at the end (just before the declaration of `ResourceMapList`)?

File: retrieval/resource_map.go [hunk 2]
```diff
@@ -77,6 +77,17 @@ var GCEResourceMap = ResourceMap{
...
 }
 
+var AnthosL4LBMap = ResourceMap{
+	Type: "anthos_l4lb",
+	LabelMap: map[string]labelTranslation{
+		ProjectIDLabel:                         constValue("project_id"),
+		KubernetesLocationLabel:                constValue("location"),
+		"__meta_kubernetes_service_annotation_anthos_l4lb_kind":          constValue("kind"),
+		"__meta_kubernetes_service_annotation_anthos_l4lb_group_name":          constValue("group_name"),
+		"__meta_kubernetes_service_annotation_anthos_l4lb_hostname":           constValue("hostname"),
```
Comment (igorpeshansky) [inline retrieval/resource_map.go:87]: Should this be run through `gofmt` again?

File: retrieval/resource_map.go (hunk 1 above)
Comment (courageJ) [reply retrieval/resource_map.go]: Moved it as advised.

File: retrieval/resource_map.go (hunk 2 above)
Comment (courageJ) [reply retrieval/resource_map.go]: Sorry I missed the step.

File: retrieval/resource_map.go [hunk 3]
```diff
@@ -131,10 +131,22 @@ var ProxyV2ResourceMap = ResourceMap{
 	},
 }
 
+var AnthosL4LBMap = ResourceMap{
```
Comment (StevenYCChou) [inline retrieval/resource_map.go:134]: Recommend adding `MatchLabel`. But it implied that you need to add the corresponding labels/annotation you specified for `MatchLabel`. An example is [ProxyV2](https://github.com/Stackdriver/stackdriver-prometheus-sidecar/blob/c8c0bfb1a5e22f5838eb6bb86608b29ef0eca0ef/retrieval/resource_map.go#L120-L132).

The motivation behind `MatchLabel` is to give option for the kubernetes users (not the sidecar maintainers) that if their `LabelMap` all match, but the users do not want to export the metrics with the monitored resource, the users just don't add the `MatchLabel` to their workload - and these metrics can be still exported as other monitored resource (e.g. `k8s_container`). 

Though I would say that the labels you set up are really specific and users are not likely to have conflict with your labels, I would still recommend adding `MatchLabel` in the first place; but we can discuss if there are other concerns.

Comment (courageJ) [reply retrieval/resource_map.go:134]: Thanks for your advice, didn't think about the edge case before. I've updated the code correspondingly.

**Comment by StevenYCChou** [comment]
LGTM. The build fails due to `coverall`, and I think it's a known issue when the branch is not from this repo. I am going to merge this PR now.

**Pull request merged**